src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,178 @@
+from typing import Union
+
+import fsspec
+from fsspec import AbstractFileSystem
+
+from unstract.sdk.exceptions import FileOperationError
+from unstract.sdk.file_storage.constants import Common, FileSeekPosition
+from unstract.sdk.file_storage.fs_interface import FileStorageInterface
+from unstract.sdk.file_storage.helper import FileStorageHelper
+
+
+class FileStorage(FileStorageInterface):
+    # This class integrates fsspec library for file operations
+
+    fs: fsspec  # fsspec file system handle
+
+    def __init__(self, provider, credentials: dict() = {}):
```

Comment:
@gaya3-zipstack do not initiate the dict like this. This will lead to unexpected issues. Keep it None as default.

Reply:
This works by reference and this variable will be only created once and basically it will be shared between different calls to this method

Reply:
Adding a sample code below to make this clear. Added list also for example. This is applicable for all data types like, dict, list , set etc .

```python
import random
import string

def test_dict(input_dict={}, input_list=[]):
    key = "".join(random.choices(string.ascii_letters, k=7))
    input_dict[key] = True
    input_list.append(key)
    return input_dict, input_list

print(test_dict())
print(test_dict())
print(test_dict())
```